entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/generator/provider/RegistrationValuesHandler.kt

```diff
@@ -0,0 +1,64 @@
+package godot.entrygenerator.generator.provider
+
+import com.squareup.kotlinpoet.ClassName
+import godot.entrygenerator.extension.assignmentPsi
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.js.resolve.diagnostics.findPsi
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.referenceExpression
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.bindingContextUtil.getReferenceTargets
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.serialization.deserialization.descriptors.DeserializedClassConstructorDescriptor
+
+abstract class RegistrationValuesHandler(
+    val propertyDescriptor: PropertyDescriptor,
+    val bindingContext: BindingContext
+) {
+    abstract fun getDefaultValueStringTemplate(): String
+    abstract fun getDefaultValueStringTemplateValues(): Array<Any>
+    abstract fun getPropertyTypeHint(): ClassName
+    abstract fun getHintString(): String
+
+    protected fun getDefaultValue(): String {
```

Comment:
Not sure if we really should have this method
the idea of `getDefaultValueStringTemplate` and `getDefaultValueStringTemplateValues` was that you could generate the template and the corresponding values to let kotlinPoet handle all imports also the ones inside the `Variant(...)`.

Reply:
I think this should be common with all supported cases, the reason why I added it as a common method in the base class is because of constructor calls. Imagine, `Vector2(1.0, 2.0)`, the check happening is recursive. Firstly it will go thru the check `expression is KtCallExpression`, then for each parameter recursively call `getDefaultValueExpression`. If we don't have this then we have de-duplicate all the checks in all registration handles. 

Ideally this function will transform `SomeCoreClass(SomeCoreClass.ENUM.VALUE, 2.0)` to `godot.core.SomeCoreClass(godot.core.SomeCoreClass.ENUM.VALUE, 2.0)`. I just need to make that change.

Reply:
I see your point but then ideally this should return a `Pair<String, Any>` with the string template as first and the values as `ClassName` or literals in second. Then KotlinPoet can handle imports. I would much more like that